arangod/StorageEngine/HotBackupFeature.cpp

```diff
@@ -23,23 +23,63 @@
 
 #include "HotBackupFeature.h"
 
+#include "Basics/FileUtils.h"
 #include "Basics/Result.h"
 #include "Cluster/ServerState.h"
 #include "Logger/Logger.h"
 #include "ProgramOptions/ProgramOptions.h"
 #include "ProgramOptions/Section.h"
+#include "IResearch/IResearchFeature.h"
+#include "RestServer/DatabaseFeature.h"
+#include "StorageEngine/EngineSelectorFeature.h"
 #include "RocksDBEngine/RocksDBEngine.h"
+#include "Scheduler/Scheduler.h"
+#include "Scheduler/SchedulerFeature.h"
 
 using namespace arangodb::application_features;
 using namespace arangodb::basics;
 using namespace arangodb::options;
 using namespace arangodb;
 
+namespace {
+
+void recreateArangoSearchViewsAfterRestore() {
+  auto* arangoSearchFeature =
+        arangodb::application_features::ApplicationServer::lookupFeature<arangodb::iresearch::IResearchFeature>("ArangoSearch");
+  LOG_TOPIC("fdeda", INFO, Logger::BACKUP)
+    << "Recreating ArangoSearch indexes...";
+  DatabaseFeature::DATABASE->enumerateDatabases(
+    [arangoSearchFeature](TRI_vocbase_t& vocbase) {
+      LOG_TOPIC("efdab", INFO, Logger::BACKUP)
+        << "Recreating ArangoSearch index for database " << vocbase.name();
+      bool res = arangoSearchFeature->recreateLocalArangoSearchData(vocbase);
```

Comment:
honestly I don't understand why do we need feature instance for this. Local helper function which drops/recreates view is more than enough.